### PR TITLE
ref: Only put un-deleted projects & teams into request.access

### PR DESCRIPTION
**Pull request by markstory**
Only load visible teams and projects into request.access. This prevents situations where a user queues a project for deletion and continues to see that project inconsistently in different parts of the application.

Both projects and teams have no cancel delete feature that would require permission checks to continue working for them.

Fixes SEN-425

### CODE REVIEW

**Comment by dcramer** [comment]
This means you wouldn’t be able to via projects which are pending delete, or if possible hit undo on delete.

**Comment by mnoble** [comment]
Makes sense to me. Everything about deleting a Project is displayed _as if_ we do it immediately, so I think this falls in line with that expectation.

**Comment by dcramer** [comment]
Im just thinking through holistically: if we did this with organizations it would break a number of checks (or require secondary checks). Not opposed to it, I just don tknow if this breaks other assumptions.

**Comment by wedamija** [comment]
I'm +1 on this too. I feel like it might be fine to have some sort of extra check going on if we want to check whether a user can access a deleted entity? Seems like the exceptional case.

**Comment by markstory** [comment]
> This means you wouldn’t be able to view projects which are pending delete, or if possible hit undo on delete.

The current project list endpoint doesn't use `access` to restrict projects, so this change won't effect the ability to get a list of deleted projects. The base ProjectEndpoint already denies all access to projects pending deletion, making undo challenging to build already.

File: src/sentry/auth/access.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 from sentry import roles
 from sentry.auth.superuser import is_active_superuser
 from sentry.models import (
-    AuthIdentity, AuthProvider, OrganizationMember, Project, SentryApp, UserPermission
+    AuthIdentity, AuthProvider, OrganizationMember, Project, SentryApp, UserPermission,
+    ProjectStatus
```
Comment (wedamija) [inline src/sentry/auth/access.py]: Nit: sort?

**Review by wedamija — APPROVED** [review]

**Pull request merged**